synthesizer/src/program/instruction/operation/macros.rs

```diff
@@ -164,6 +164,116 @@ macro_rules! operation {
             }
         }
     };
+    // K-ary operation with at least 1 output.
+    ($vis:vis struct $name:ident<$operator:path, $circuit_operator:path, $operate:ident, $opcode:tt, $num_inputs:tt, $num_outputs:tt> { $( ( $($input:ident),+ ) => ( $($output:ident),+ ) $( ($($condition:tt),+) )?, )+ }) => {
+        /// The implementation of the operation.
+        #[derive(Clone, PartialEq, Eq, Hash)]
```

Comment:
are the equality and hash impls meaningful here? the `struct` has no contents after all, so hashing is a noop, and equality checks always return `true`